--- origsrc/gst-plugins-bad-0.10.22/configure.ac	2011-12-14 19:59:26.349792800 -0600
+++ src/gst-plugins-bad-0.10.22/configure.ac	2011-12-14 20:38:41.482498700 -0600
@@ -1042,6 +1042,9 @@ PKG_CHECK_EXISTS(mjpegtools >= 1.6.1.93
   PKG_CHECK_EXISTS(mjpegtools >= 1.8.0 mjpegtools < 1.9.0, [
     mjpegtools_api=10800
   ], [
+  PKG_CHECK_EXISTS(mjpegtools >= 2.0.0, [
+    mjpegtools_api=20000
+  ], [
     PKG_CHECK_MODULES(MJPEG, mjpegtools >= 1.9.0, [
       dnl logging API changed in release candidates
       OLD_CFLAGS="$CFLAGS"
@@ -1059,6 +1062,7 @@ PKG_CHECK_EXISTS(mjpegtools >= 1.6.1.93
       mjpegtools_api=0
     ])
   ])
+  ])
 ])
 AC_DEFINE_UNQUOTED(GST_MJPEGTOOLS_API, $mjpegtools_api,
   [mjpegtools API evolution])
@@ -1147,10 +1151,14 @@ AG_GST_CHECK_FEATURE(MPLEX, [mplex], mpl
     dnl and mpeg2enc, we check for mplex on its own, too.
     dnl libmplex < 1.9rc? has fuzzy ABI, valgrind and other problems
     PKG_CHECK_MODULES(MPLEX, mjpegtools >= 1.9.0, [
+      dnl HACK because mplex 2.0.0 header files have a spurious 'include config.h'
+      touch config.h
       dnl switch over to c++ to test things
       AC_LANG_CPLUSPLUS
       OLD_CPPFLAGS="$CPPFLAGS"
-      CPPFLAGS="$CPPFLAGS $MPLEX_CFLAGS"
+      dnl HACK as above
+      CPPFLAGS_GOOD="$CPPFLAGS $MPLEX_CFLAGS"
+      CPPFLAGS="$CPPFLAGS_GOOD -I."
       AC_CHECK_HEADER(interact.hpp, [
         MPLEX_LIBS="$MPLEX_LIBS -lmplex2 $LIBM"
         OLD_LIBS="$LIBS"
@@ -1187,6 +1195,7 @@ main (int   argc,
 }
         ],[
           HAVE_MPLEX="yes"
+          CPPFLAGS="$CPPFLAGS_GOOD"
           AC_SUBST(MPLEX_CFLAGS)
           AC_SUBST(MPLEX_LIBS)
           AC_SUBST(MPLEX_LDFLAGS)
--- origsrc/gst-plugins-bad-0.10.22/ext/mpeg2enc/gstmpeg2encoder.cc	2010-04-29 03:59:29.000000000 -0500
+++ src/gst-plugins-bad-0.10.22/ext/mpeg2enc/gstmpeg2encoder.cc	2011-12-14 20:28:41.360173700 -0600
@@ -26,7 +26,10 @@
 
 #include <mpegconsts.h>
 #include <quantize.hh>
-#if GST_MJPEGTOOLS_API >= 10900
+#if GST_MJPEGTOOLS_API >= 20000
+#include <ontheflyratectlpass1.hh>
+#include <ontheflyratectlpass2.hh>
+#elif GST_MJPEGTOOLS_API >= 10900
 #include <ontheflyratectl.hh>
 #include <pass1ratectl.hh>
 #include <pass2ratectl.hh>
