--- origsrc/gst-plugins-bad-0.10.7/ext/mpeg2enc/gstmpeg2encstreamwriter.cc	2008-03-04 23:38:06.000000000 -0600
+++ src/gst-plugins-bad-0.10.7/ext/mpeg2enc/gstmpeg2encstreamwriter.cc	2008-07-02 14:49:56.515625000 -0500
@@ -48,8 +48,8 @@
 }
 
 void
-GstMpeg2EncStreamWriter::WriteOutBufferUpto (const guint8 * buffer,
-    const guint32 flush_upto)
+GstMpeg2EncStreamWriter::WriteOutBufferUpto (const uint8_t * buffer,
+    const uint32_t flush_upto)
 {
   GstBuffer *buf, *inbuf;
   GstMpeg2enc *enc = GST_MPEG2ENC (GST_PAD_PARENT (pad));
--- origsrc/gst-plugins-bad-0.10.7/ext/mpeg2enc/gstmpeg2encstreamwriter.hh	2008-03-04 23:38:06.000000000 -0600
+++ src/gst-plugins-bad-0.10.7/ext/mpeg2enc/gstmpeg2encstreamwriter.hh	2008-07-02 14:49:56.531250000 -0500
@@ -35,8 +35,8 @@
   ~GstMpeg2EncStreamWriter ();
 
   /* output functions */
-  void WriteOutBufferUpto (const guint8 * buffer,
-      const guint32 flush_upto);
+  void WriteOutBufferUpto (const uint8_t * buffer,
+      const uint32_t flush_upto);
   guint64 BitCount ();
 
   private:
--- origsrc/gst-plugins-bad-0.10.7/gst/librfb/d3des.h	2007-10-31 09:09:25.000000000 -0500
+++ src/gst-plugins-bad-0.10.7/gst/librfb/d3des.h	2008-07-02 14:49:56.531250000 -0500
@@ -22,7 +22,7 @@
 #define EN0	0               /* MODE == encrypt */
 #define DE1	1               /* MODE == decrypt */
 
-extern void deskey (unsigned char *, int);
+extern void deskey (unsigned char *, int32_t);
 
 /*		      hexkey[8]     MODE
  * Sets the internal key register according to the hexadecimal
--- origsrc/gst-plugins-bad-0.10.7/gst/librfb/vncauth.h	2007-10-31 09:09:25.000000000 -0500
+++ src/gst-plugins-bad-0.10.7/gst/librfb/vncauth.h	2008-07-02 14:49:56.546875000 -0500
@@ -24,7 +24,7 @@
 #define MAXPWLEN 8
 #define CHALLENGESIZE 16
 
-extern int vncEncryptAndStorePasswd (char *passwd, char *fname);
+extern int32_t vncEncryptAndStorePasswd (char *passwd, char *fname);
 extern char *vncDecryptPasswdFromFile (char *fname);
 extern void vncRandomBytes (unsigned char *bytes);
 extern void vncEncryptBytes (unsigned char *bytes, char *passwd);
